File: app/src/main/java/dev/blank/topheadline/viewmodel/NewsViewModelFactory.java
package dev.blank.topheadline.viewmodel;

import android.app.Application;

import androidx.annotation.NonNull;
import androidx.lifecycle.ViewModel;
import androidx.lifecycle.ViewModelProvider;

public class NewsViewModelFactory implements ViewModelProvider.Factory {
    private Application mApplication;

    public NewsViewModelFactory(Application application) {
        mApplication = application;
    }

    @NonNull
    @Override
    public <T extends ViewModel> T create(@NonNull Class<T> modelClass) {
        return (T) new NewsViewModel(mApplication);
    }
}

File: app/src/main/java/dev/blank/topheadline/viewmodel/NewsViewModelFactory.kt
package dev.blank.topheadline.viewmodel

import android.app.Application
import androidx.lifecycle.ViewModel
import androidx.lifecycle.ViewModelProvider

class NewsViewModelFactory(private val mApplication: Application) : ViewModelProvider.Factory {
    override fun <T : ViewModel?> create(modelClass: Class<T>): T {
        return NewsViewModel(mApplication) as T
    }

}